catalog/transformer/catalog-transformer-tika-input/src/main/java/ddf/catalog/transformer/input/tika/TikaInputTransformer.java

```diff
@@ -410,53 +455,62 @@ public Metacard transform(InputStream input, String id)
+
             if (textContentHandler != null) {
                 String plainText = textContentHandler.toString();
-
-                Set<AttributeDescriptor> attributes = contentMetadataExtractors.values()
-                        .stream()
-                        .map(ContentMetadataExtractor::getMetacardAttributes)
-                        .flatMap(Collection::stream)
-                        .collect(Collectors.toSet());
-                MetacardTypeImpl extendedMetacardType = new MetacardTypeImpl(metacardType.getName(),
-                        metacardType,
-                        attributes);
-
-                metacard = MetacardCreator.createMetacard(metadata,
-                        id,
-                        metadataText,
-                        extendedMetacardType);
-
-                for (ContentMetadataExtractor contentMetadataExtractor : contentMetadataExtractors.values()) {
+                for (ContentMetadataExtractor contentMetadataExtractor : contentExtractors.values()) {
                     contentMetadataExtractor.process(plainText, metacard);
                 }
-            } else {
-                metacard = MetacardCreator.createMetacard(metadata, id, metadataText, metacardType);
             }
 
-            if (StringUtils.isNotBlank(metacardContentType)) {
-                metacard.setAttribute(new AttributeImpl(Core.DATATYPE,
-                        getDatatype(metacardContentType)));
+            for (MetadataExtractor metadataExtractor : metadataExtractors.values()) {
+                metadataExtractor.process(metadataText, metacard);
             }
 
-            if (StringUtils.startsWith(metacardContentType, "image")) {
-                try (InputStream inputStreamCopy = fileBackedOutputStream.asByteSource()
-                        .openStream()) {
-                    createThumbnail(inputStreamCopy, metacard);
-                }
-            }
+            enrichMetacard(fileBackedOutputStream, metacardContentType, bytes, metacard);
 
             LOGGER.debug("Finished transforming input stream using Tika.");
             return metacard;
         }
     }
 
+    protected void enrichMetacard(TemporaryFileBackedOutputStream fileBackedOutputStream,
+            String metacardContentType, long bytes, Metacard metacard) throws IOException {
+
+        if (StringUtils.isNotBlank(metacardContentType)) {
+            metacard.setAttribute(new AttributeImpl(Core.DATATYPE,
+                    getDatatype(metacardContentType)));
+        }
+
+        if (StringUtils.startsWith(metacardContentType, "image")) {
+            try (InputStream inputStreamCopy = fileBackedOutputStream.asByteSource()
+                    .openStream()) {
+                createThumbnail(inputStreamCopy, metacard);
+            }
+        }
+
+        metacard.setAttribute(new AttributeImpl(Core.RESOURCE_SIZE, String.valueOf(bytes)));
+
+    }
+
+    protected Set<AttributeDescriptor> getAllExtractorAttributes() {
```

Comment:
Helper method to gather call the attribute descriptors from all the `ContentMetadataExtractors` and `MetadataExtractors`.